dapr-spring/dapr-spring-boot-autoconfigure/src/main/java/io/dapr/spring/boot/autoconfigure/client/DaprClientAutoConfiguration.java

```diff
@@ -15,34 +15,37 @@
 
 import io.dapr.client.DaprClient;
 import io.dapr.client.DaprClientBuilder;
+import io.dapr.config.Properties;
 import io.dapr.spring.core.client.DaprClientCustomizer;
 import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 
 import java.util.stream.Collectors;
 
 @AutoConfiguration
 @ConditionalOnClass(DaprClient.class)
+@EnableConfigurationProperties(DaprClientProperties.class)
 public class DaprClientAutoConfiguration {
 
   @Bean
-  @ConditionalOnMissingBean
-  DaprClientBuilderConfigurer daprClientBuilderConfigurer(ObjectProvider<DaprClientCustomizer> customizerProvider) {
-    DaprClientBuilderConfigurer configurer = new DaprClientBuilderConfigurer();
-    configurer.setDaprClientCustomizer(customizerProvider.orderedStream().collect(Collectors.toList()));
-
-    return configurer;
+  @ConditionalOnMissingBean(DaprConnectionDetails.class)
+  DaprConnectionDetails daprConnectionDetails(DaprClientProperties properties) {
+    return new PropertiesDaprConnectionDetails(properties);
   }
 
   @Bean
   @ConditionalOnMissingBean
-  DaprClientBuilder daprClientBuilder(DaprClientBuilderConfigurer daprClientBuilderConfigurer) {
+  DaprClientBuilder daprClientBuilder(DaprClientProperties daprClientProperties) {
```

Comment:
should use `DaprConnectionDetails` because the implementation can change from `PropertiesDaprConnectionDetails` to the one provided by `DaprContainerConnectionDetailsFactory`

```suggestion
  DaprClientBuilder daprClientBuilder(DaprConnectionDetails daprConnectionDetails) {
```